test/framework/integration_framework/integration_test_framework.hpp

```diff
@@ -60,15 +61,23 @@ namespace integration_framework {
      */
     explicit IntegrationTestFramework(
         size_t maximum_proposal_size = 10,
-        void (*destructor_lambda)(IntegrationTestFramework *) = nullptr)
-        : maximum_proposal_size_(maximum_proposal_size),
-          destructor_lambda_(destructor_lambda) {}
+        const std::function<void(IntegrationTestFramework *)> &deleter =
```

Comment:
What is the reason to pass the pointer? Maybe rework it with a reference?

Reply:
That may cause an ub, you pass reference to `delete`